File: frontend/src/stores/pet.ts
import { create } from 'zustand'

interface Pet {
  id: number
  name: string
  pet_type: string
  stage: string
  level: number
  exp: number
  hunger: number
  happiness: number
  energy: number
}

interface PetState {
  pet: Pet | null
  setPet: (pet: Pet) => void
  updateStats: (stats: Partial<Pet>) => void
}

export const usePetStore = create<PetState>((set) => ({
  pet: null,
  setPet: (pet) => set({ pet }),
  updateStats: (stats) => set((state) => ({
    pet: state.pet ? { ...state.pet, ...stats } : null
  }))
}))

File: backend/app/schemas/pet.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional
from ..models.models import PetType, PetStage

class PetCreate(BaseModel):
    name: str
    pet_type: PetType

class PetUpdate(BaseModel):
    name: Optional[str] = None

class PetOut(BaseModel):
    id: int
    name: str
    pet_type: PetType
    stage: PetStage
    level: int
    exp: int
    hunger: float
    happiness: float
    energy: float
    last_fed: datetime
    last_played: datetime
    
    class Config:
        from_attributes = True

class FeedRequest(BaseModel):
    item_id: int

class PlayRequest(BaseModel):
    duration: int = 10  # minutes
